lib/plugins/object.js

```diff
@@ -194,6 +195,77 @@ module.exports = fp(
       }
     );
 
+    fastify.decorate(
+      'checkAndFixCorruptedHead',
+      /**
+       * Properly ensure the object head is correctly
+       * pointing to existing object versions and
+       * fix it otherwise.
+       *
+       * @param {Object} opts Method parameters
+       * @param {string} opts.name Object name
+       * @param {string} opts.env Environment
+       * @returns {Promise<boolean>} Return true if head was corrupted, false otherwise
+       */
+      async function checkAndFixCorruptededHead({ name, env }) {
+        const [obj, objVariants] = await Promise.all([
+          fastify.getObject({ name, env }),
+          fastify.getAllObjectVersions({ name, env })
+        ]);
+
+        if (objVariants.length === 0) {
+          // Delete object version
+          await fastify.deleteObject({ name, env });
+          return true;
+        }
+
+        const { headVersion, latestVersion } = obj;
+
+        const newHeadVersion = objVariants.includes(headVersion)
+          ? headVersion
+          : null;
+
+        let newLatestVersion;
+        if (!objVariants.includes(latestVersion)) {
+          // Get the new latest available version
+          newLatestVersion = objVariants.sort(compareVersions)[
+            objVariants.length - 1
+          ];
+        } else {
+          newLatestVersion = latestVersion;
+        }
+
+        // Update db record if head was corrupted
+        if (
+          newHeadVersion !== headVersion ||
+          newLatestVersion !== latestVersion
+        ) {
+          await dynamo
+            .update({
+              TableName: OBJECTS_TABLE,
+              Key: { keyname: name, env },
+              UpdateExpression: 'set #a = :x, #b = :z',
+              ConditionExpression: '#a = :i and #b = :y',
+              ExpressionAttributeNames: {
+                '#a': 'headVersion',
+                '#b': 'latestVersion'
+              },
+              ExpressionAttributeValues: {
+                ':x': newHeadVersion,
+                ':z': newLatestVersion,
+                ':i': headVersion,
+                ':y': latestVersion
```

Comment:
Do these vars here and in values need to be single-character? Or can we use more meaningful names?

```suggestion
              UpdateExpression: 'set #headVersion = :newHeadVersion, #latestVersion = :newLatestVersion',
              ConditionExpression: '#headVersion = :headVersion and #latestVersion = :latestVersion',
              ExpressionAttributeNames: {
                '#headVersion': 'headVersion',
                '#latestVersion': 'latestVersion'
              },
              ExpressionAttributeValues: {
                ':newHeadVersion': newHeadVersion,
                ':newLatestVersion': newLatestVersion,
                ':headVersion': headVersion,
                ':latestVersion': latestVersion
```

Reply:
The point of using single letters is to then benefit of `ExpressionAttributeNames` and `ExpressionAttributeValues` engine replacement. The suggestion you propose doesn't make much sense. In that case we could simply use the explicit name within the query string itself.

Reply:
FYI, I am not against in not using `ExpressionAttributeNames`. I personally prefer the shorter syntax though.

Reply:
Updated this by the way. I made things more explicit and removed `ExpressionAttributeNames`.